Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1639,38 +1759,39 @@ void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizes(GridLayoutState& gridL
 
 void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizesMasonry(GridLayoutState& gridLayoutState)
 {
-    if (m_strategy->isComputingSizeContainment()) {
+    if (m_strategy->isComputingSizeContainment() || !m_grid.hasGridItems()) {
         handleInfinityGrowthLimit();
         return;
     }
+    StdMap<SpanLength, MasonryMinMaxTrackSize> indefiniteItems;
+    StdMap<SpanLength, Vector<MasonryMinMaxTrackSizeWithGridSpan>> definiteItems;
```

Comment:
We should probably indicate that these are indefinite items spanning flex tracks, as opposed to its counterpart, so maybe something like `indefiniteItemsSpanningFlexTracks`?

Reply:
That is not quite correct. `indefiniteItems` does not care whether we span flex tracks or not. When we "create fake grid items" in `convertIndefiniteItemsToDefiniteMasonry` we start to care about whether it spans or flex tracks or not, but the results of that action are stored in the definite items.

Reply:
Oops I meant the `definiteItems` structure. Does that still hold up?